File: leetcode3/최원준/820. Short Encoding of Words.py
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        words = sorted(set(words), key=lambda word: word[::-1])

        answer = 0

        for i, word in enumerate(words):
            if i + 1 < len(words) and words[i + 1].endswith(word):
                continue

            answer += len(word) + 1

        return answer

File: leetcode3/강병호/820_Short Encoding of Words.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int minimumLengthEncoding(String[] words) {
        Set<String> goodWords = new HashSet<>(Arrays.asList(words));
        
        for (String word : words) {
            for (int k = 1; k < word.length(); ++k) {
                goodWords.remove(word.substring(k));
            }
        }
        
        int totalLength = 0;
        for (String word : goodWords) {
            totalLength += word.length() + 1;
        }
        
        return totalLength;
    }
}